sycl/source/device_selector.cpp

```diff
@@ -199,6 +192,12 @@ __SYCL_EXPORT int accelerator_selector_v(const device &dev) {
   return Score;
 }
 
+int host_selector::operator()(const device &dev) const {
```

Comment:
can we just remove the whole `host_selector` class?

Reply:
Arguably it makes sense that the `host_selector` would reject all devices when we know there is no host available, as it is no different than using `gpu_selector` on a system without a GPU. However, I do somewhat agree that there could easily be users that are expecting it to _just work_ and having it throw after this. However, it seems like two somewhat equal evils, so I would like to stick with the deprecation as it follows our typical strategy. Hopefully users will heed the deprecation warning!

Reply:
I'd vote for removing as much as possible, but I won't insist.